tests/user_examples/extra_src/shared/general_continuum/continuum_dynamics/continuum_dynamics_complex.hpp

```diff
@@ -70,5 +70,125 @@ void BaseShearStressRelaxation2ndHalfWithWall<BaseShearStressRelaxation2ndHalfTy
+        for (size_t n = 0; n != wall_neighborhood.current_size_; ++n)
+        {
+            size_t index_j = wall_neighborhood.j_[n];
+            Vecd& e_ij = wall_neighborhood.e_ij_[n];
+            Real dW_ijV_j = wall_neighborhood.dW_ijV_j_[n];
+            Real r_ij = wall_neighborhood.r_ij_[n];
+            Vecd nablaW_ijV_j = wall_neighborhood.dW_ijV_j_[n] * wall_neighborhood.e_ij_[n];
+
+            Real face_wall_external_acceleration = (acc_prior_i - acc_ave_k[index_j]).dot(-e_ij);
+            Real p_in_wall = this->p_[index_i] + this->rho_[index_i] * r_ij * SMAX(Real(0), face_wall_external_acceleration);
+            Matd stress_tensor_in_wall = stress_tensor_i;
+            acceleration += (stress_tensor_i + stress_tensor_in_wall) * nablaW_ijV_j;
+            rho_dissipation += this->riemann_solver_.DissipativeUJump(this->p_[index_i] - p_in_wall) * dW_ijV_j;
+        }
+    }
+    this->acc_[index_i] += acceleration / this->rho_[index_i];
+    this->drho_dt_[index_i] += rho_dissipation * this->rho_[index_i];
+}
+//=================================================================================================//
+template <class BaseStressRelaxation2ndHalfType>
+void BaseStressRelaxation2ndHalfWithWall<BaseStressRelaxation2ndHalfType>::interaction(size_t index_i, Real dt)
+{
+    BaseStressRelaxation2ndHalfType::interaction(index_i, dt);
+
+    Real density_change_rate = 0.0;
+    Vecd p_dissipation = Vecd::Zero();
+
+    Vecd vel_i = this->vel_[index_i];
+    Matd velocity_gradient = Matd::Zero();
+
+    for (size_t k = 0; k < fluid_dynamics::FluidWallData::contact_configuration_.size(); ++k)
+    {
+        StdLargeVec<Vecd>& vel_ave_k = *(this->wall_vel_ave_[k]);
+        StdLargeVec<Vecd>& n_k = *(this->wall_n_[k]);
+
+        Neighborhood& wall_neighborhood = (*fluid_dynamics::FluidWallData::contact_configuration_[k])[index_i];
+        for (size_t n = 0; n != wall_neighborhood.current_size_; ++n)
+        {
+            size_t index_j = wall_neighborhood.j_[n];
+            Vecd& e_ij = wall_neighborhood.e_ij_[n];
+            Real dW_ijV_j = wall_neighborhood.dW_ijV_j_[n];
+            Vecd nablaW_ijV_j = wall_neighborhood.dW_ijV_j_[n] * wall_neighborhood.e_ij_[n];
+
+            Real beta = 1.9;
+            Vecd vel_in_wall = (1 - beta) * vel_i + beta * vel_ave_k[index_j];
+
+            Matd velocity_gradient_ij = -(vel_i - vel_in_wall) * nablaW_ijV_j.transpose();
+            velocity_gradient += velocity_gradient_ij;
+
+            density_change_rate += (this->vel_[index_i] - vel_in_wall).dot(e_ij) * dW_ijV_j;
+            Vecd u_jump = 2.0 * (this->vel_[index_i] - vel_ave_k[index_j]);
+            p_dissipation += this->riemann_solver_.DissipativePJumpExtra(u_jump, n_k[index_j]) * dW_ijV_j;
+        }
+    }
+    this->drho_dt_[index_i] += density_change_rate * this->rho_[index_i];
+    this->velocity_gradient_[index_i] += velocity_gradient;
+    this->acc_[index_i] += p_dissipation / this->rho_[index_i];
```

Comment:
Try  to remove the introduction of some once-used temporal variables, viz. a in " a =b , c =a * f" , directly use " c = b * f"